PetrolEngine.cpp

```diff
@@ -10,11 +11,49 @@ PetrolEngine::PetrolEngine(int power, float capacity, int gears)
     std::cout << __FUNCTION__ << std::endl;
 }
 
+PetrolEngine::~PetrolEngine()
+{
+    std::cout << __FUNCTION__ << std::endl;
+}
+
 void PetrolEngine::changeGear(int gear)
 {
     // TODO: Add checking if gear is between -1 and gears_
     // -1 is for REAR
     // 0 is for NEUTRAL
-    currentGear_ = gear;
-    std::cout << __FUNCTION__ << std::endl;
+    if (isGearValid(gear))
+    {
+	try
+	{
+	    if (currentGear_ > 0 && gear == -1)
+	    {
+		throw InvalidGear();
+	    }
+	    else
+	    {
+		currentGear_ = gear;
+		std::cout << "Gear has been changed to: " << currentGear_ << std::endl;
+	    }
+
+	}
+	catch (const InvalidGear& e)
+	{
+	    std::cout << "Bad function call." << e.what() << std::endl;
+	}
+    	std::cout << __FUNCTION__ << std::endl;
+    }
+    else
+    {
+	std::cout<<"This gear is not valid." <<std::endl;
+    }
+}
```

Comment:
ta funkcja jakos za bardzo nie mowi czy ten gear sie ustawil czy jednak nie.
jak odroznic "OK" i "NOK" case?

Reply:
Po outpucie not valid, mozna sie domyslic ze bieg sie nie wrzucil, ale masz racje lepiej powiedziec wprost, ze cos jest biale, a nie mowic ze na pewno nie jest czarne :)

Reply:
Poprawiony na: This gear is not valid and hasn't changed.

Reply:
chodzilo mi bardziej o kod xd
np jesli mam 10 gearow do wyboru i bedzie tam tylko 1 jaki zadziala. Mam wtedy czytac konsole? xd
A jesli ten soft bedzie dzialac w Tesli to kto ma to czytac?